Contest/Assignment/src/org/togetherjava/event/elevator/humans/Human.java

```diff
@@ -17,12 +21,13 @@ public final class Human implements ElevatorListener {
     private State currentState;
     private final int startingFloor;
     private final int destinationFloor;
+    private Elevator bestElevator = null;
     /**
      * If the human is currently inside an elevator, this is its unique ID.
      * Otherwise, this is {@code null} to indicate that the human is currently on the corridor.
      */
     private Integer currentEnteredElevatorId;
-
+    private final List<HumanArrivedListener> listeners = new ArrayList<>();
```

Comment:
design-wise this is a bit weird. like, why would humans contact anyone when they arrived. more reasonable might be something like an elevator notifying the elevator system when a human leaves the elevator.
but even that is technically IRL not a thing - anyways, fair solution to the performance problem.